Automattic/jetpack
# WhatsApp block: Increase specificity for background styling

Fixes #18688

#### Changes proposed in this Pull Request:

* When the WhatsApp button is focused on the frontend, the icon disappears. This is caused by a style from the Twenty Twenty One theme applying a `background` style that overrides the block style when focused. This PR adds a selector with higher specificity to make sure the background styling is not overridden.
* Also a small tweak to the editor styles to correctly center the button text and remove unnecessary bottom border in some themes.

**Notes**
Adding the `:focus` selector was not enough to increase the specificity; I also had to select on the parent `div.wp-block-jetpack-whatsapp-button`. I added this in a separate selector rather than applying to all of the WhatsApp button styles to prevent the styling from breaking if the user deletes the parent block in the code editor.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Sync this branch.
* Activate the Twenty Twenty One theme.
* Create a new post and insert a WhatsApp button block.
* View the post on the frontend.
* Click on the button and observe that the icon does not disappear.
* Clicking will redirect the page, so you can also test by forcing the ":focus" state in the Chrome devtools or Tabbing to the component in order to focus it.

**Before**
<img width="283" alt="Screen Shot 2021-03-30 at 4 16 59 PM" src="https://user-images.githubusercontent.com/63313398/113068623-7da27f00-9173-11eb-89a8-d493f667156b.png">

**After**
<img width="263" alt="Screen Shot 2021-03-30 at 4 17 29 PM" src="https://user-images.githubusercontent.com/63313398/113068652-901cb880-9173-11eb-90d5-61bed469cf98.png">
